backend/fms_core/services/derived_sample.py

```diff
@@ -14,4 +14,23 @@ def inherit_derived_sample(derived_sample_source, new_derived_sample_data):
     except Error as e:
         errors.append(';'.join(e.messages))
     
-    return (new_derived_sample, errors, warnings)
\ No newline at end of file
+    return (new_derived_sample, errors, warnings)
+
+def update_qc_flags(sample, quantity_flag, quality_flag):
+    errors = []
+    warnings = []
+
+    try:
+        derived_sample = sample.derived_sample_not_pool
```

Comment:
This is true so far but will it still be true for library QC ... If you have a service for updating the QC flags you must expect to reuse it ... Maybe you should move the service to sample.py since you pass a sample along ... or to full_sample.py...
I see that function as a facilitator that will set all derived samples attached to a sample to the values given...